test/e2e/nodeobservability_test.go

```diff
@@ -134,13 +142,122 @@ var _ = Describe("Node Observability Operator end-to-end test suite", Ordered, f
+				Labels: map[string]string{
+					hostnameLabel: nodesWithoutOperator.Items[0].Labels[hostnameLabel],
+				},
+				Type: operatorv1alpha1.CrioKubeletNodeObservabilityType,
+			},
+		}
+		By("deploying Node Observability Agents", func() {
+			Expect(k8sClient.Create(ctx, nodeobservability)).To(Succeed(), "NodeObservability resource is expected to be created")
+			Eventually(func() bool {
+				ds := &appsv1.DaemonSet{}
+				dsNamespacedName := types.NamespacedName{
+					Name:      "node-observability-ds",
+					Namespace: operatorNamespace,
+				}
+				Expect(client.IgnoreNotFound(k8sClient.Get(ctx, dsNamespacedName, ds))).To(Succeed())
+				return ds.Status.NumberReady != 0 && ds.Status.DesiredNumberScheduled == ds.Status.NumberReady
+			}, 60, time.Second).Should(BeTrue(), "number of ready agents != number of desired agents")
+		})
+	})
+	Context("Happy Path scenario - single scrape is initiated and it is expected to succeed", func() {
+		var (
+			nodeobservabilityRun *operatorv1alpha1.NodeObservabilityRun
+		)
+		BeforeEach(func() {
+			nodeobservabilityRun = &operatorv1alpha1.NodeObservabilityRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      defaultTestName,
+					Namespace: operatorNamespace,
+				},
+				Spec: operatorv1alpha1.NodeObservabilityRunSpec{
+					NodeObservabilityRef: &operatorv1alpha1.NodeObservabilityRef{
+						Name: "cluster",
+					},
+				},
+			}
+		})
+
+		It("runs Node Observability scrape", func() {
+
+			By("by initiating the scrape", func() {
+				Expect(k8sClient.Create(ctx, nodeobservabilityRun)).To(Succeed(), "NodeObservabilityRun resource is expected to be created")
+				GinkgoWriter.Println("Creation of NOB run done")
+			})
+
+			By("by collecting status", func() {
+				run := &operatorv1alpha1.NodeObservabilityRun{}
+				runNamespacedName := types.NamespacedName{
+					Name:      defaultTestName,
+					Namespace: operatorNamespace,
+				}
+				Eventually(func() bool {
+					Expect(k8sClient.Get(ctx, runNamespacedName, run)).To(Succeed(), "NOB run is expected to exist")
+					return run.Status.FinishedTimestamp.IsZero()
+				}, 600, time.Second).Should(BeFalse())
+				Expect(run.Status.FailedAgents).To(BeEmpty(), "Failed agent list is expected to be empty")
+			})
+		})
+
+		AfterEach(func() {
+			Expect(k8sClient.Delete(ctx, nodeobservabilityRun)).To(Succeed())
+		})
+	})
+
 	AfterAll(func() {
-		Expect(k8sClient.Delete(ctx, nodeobservability)).To(Succeed())
+		if !CurrentSpecReport().Failed() {
+			GinkgoWriter.Println("Cleaning up NOB")
+			Expect(k8sClient.Delete(ctx, nodeobservability)).To(Succeed())
+			GinkgoWriter.Println("Waiting for NOBMC to disappear")
+			nobmc := &operatorv1alpha1.NodeObservabilityMachineConfig{}
+			Eventually(func() bool {
+				return errors.IsNotFound(k8sClient.Get(ctx, types.NamespacedName{Name: "cluster"}, nobmc))
+			}, 600, time.Second).Should(BeTrue())
+		} else {
+			// keep NOB CR to preserve all the dependent objects
```

Comment:
I was thinking of doing this by running a must-gather within out test job. But I guess this works as well.

Reply:
It worked kinda OK when was run on local: allowed to keep NodeObservability CRs and all owned objects. In the CI, it's a bit less useful as `must-gather` [doesn't seem to collect non CVO resources](https://github.com/openshift/must-gather#collection-scripts). However `MCP` and `MC` which are owned by `NOBMC` get saved in CI too. In this regard I'm thinking whether running `must-gather` from the job would bring any more data than the "standard" must-gather done after the test.

Reply:
I was thinking of just re-using steps that upload the must-gathers as artifacts and providing must-gather a custom collection script from our operator repo (we can pull all the resources we need collected).